src/app/home-page/widget/organization-box/organization-box.component.ts

```diff
@@ -38,45 +39,22 @@ export class OrganizationBoxComponent extends Base implements OnInit {
 
   ngOnInit(): void {
     this.getMyOrganizations();
-    this.requestsQuery.myPendingOrganizationRequests$.subscribe(
-      (requests) => {
-        if (requests) {
-          this.pendingRequests = requests;
-        }
-      },
-      (error: HttpErrorResponse) => {
-        this.alertService.detailedError(error);
-      }
-    );
-    this.requestsQuery.myOrganizationInvites$.subscribe(
-      (invites) => {
-        if (invites) {
-          this.pendingRequests = invites;
-        }
-      },
-      (error: HttpErrorResponse) => {
-        this.alertService.detailedError(error);
-      }
-    );
+    this.pendingRequests$ = this.requestsQuery.myPendingOrganizationRequests$;
+    this.pendingInvites$ = this.requestsQuery.myOrganizationInvites$;
   }
 
   private getMyOrganizations() {
     this.usersService
-      .getUserDockstoreOrganizations()
+      .getUserDockstoreOrganizations(null, this.filterText)
       .pipe(
         finalize(() => (this.isLoading = false)),
         takeUntil(this.ngUnsubscribe)
       )
       .subscribe(
         (myOrgs: Array<OrganizationUpdateTime>) => {
-          this.listOfOrganizations = [];
-          myOrgs.forEach((org: OrganizationUpdateTime) => {
-            if (this.listOfOrganizations.length < 5) {
-              this.listOfOrganizations.push(org);
-            }
-            if (this.firstCall) {
-              this.totalOrgs += 1;
-            }
+          myOrgs.forEach(() => {
+            this.listOfOrganizations = myOrgs.slice(0, 7);
+            this.totalOrgs = myOrgs.length;
           });
         },
         (error: HttpErrorResponse) => {
```

Comment:
It was like this, but you could set `this.listOfOrganizations = []` in the error block to handle this case:

1. User's organizations are successfully fetched
2. User types in some text and there's an error fetching the organizations, e.g., the network connection is lost, web service fails, etc.
3. We display the snackbar message, but the organizations from step are still displayed.

You could argue it's more user-friendly to keep the old orgs displayed, and maybe it is. On the other hand, our error messages aren't very prominent and the user might be surprised to see orgs that don't match the filter. So your call :)